### DISCUSSION TITLE
how to make dropdown variables in a dashboard?

### DISCUSSION THREAD

**Discussion by farmboy-dev** [grafana/grafana-infinity-datasource#580] [Q&A]
![image](https://user-images.githubusercontent.com/70833564/232138509-fa011865-e0c1-4931-9f0f-b6d352373dc8.png)
I made a variable using graphql and uql, but not working.
{
  organizations (first:100){
      nodes{
        name
      }
  }
}
this is my graphql query, and I want to list organizations' name. 
![image](https://user-images.githubusercontent.com/70833564/232138997-d84433ec-9216-4ff3-9091-197f4aeaeae9.png)
when I test this using explore, it can show list well.
 
![image](https://user-images.githubusercontent.com/70833564/232139149-7d4b21d9-8f04-488c-836e-8064fa9fb9e4.png)

Summary, I want to make a dropdown list using GitHub organization name. 
Thank you in advance.

**Comment by farmboy-dev** [comment]
![image](https://user-images.githubusercontent.com/70833564/232147445-2155a539-297a-4bf7-91f0-f9118aa262e7.png)
![image](https://user-images.githubusercontent.com/70833564/232147475-58505fa8-35b6-4a83-8d14-f6f80aa76b71.png)
when I added inline data and tried Json type query, it works.
What is the difference? 
Thank you

**Comment by aakashH242** [comment]
From your screenshots, it seems you are not actually querying the Github GraphQL API but rather, you're using your query as the data for Infinity. Please note that the "Source-Inline" option that you have selected actually sets the data Infinity will be using to whatever you provide in the data field.
You will instead need to change the Source to URL, put in your Github endpoint and pass in the query as part of request parameters.
Please check the github Graph QL API documentation and how it expects the query request.
For example, Cloudflare Graph QL API accepts a POST request with the query in the request body as a JSON.
```
Body: {"query": "{...}"}
```
Then, you can configure Infinity in your variable to call the github graph QL API with the appropriate request method and body.
